internal/util/target.go

```diff
@@ -90,6 +91,36 @@ func shootForTargetViaSeed(ctx context.Context, gardenClient client.Client, t ta
 	return matchingShoots[0], nil
 }
 
+// ShootsForTarget returns all possible shoots for a given target. The
+// target must either target a project or a seed (both including a garden).
+func ShootsForTarget(ctx context.Context, gardenClient client.Client, t target.Target) ([]gardencorev1beta1.Shoot, error) {
+	var listOpt client.ListOption
+
+	if t.ProjectName() != "" {
+		project, err := ProjectForTarget(ctx, gardenClient, t)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch project: %w", err)
+		}
+
+		if project.Spec.Namespace == nil {
+			return nil, nil
+		}
+
+		listOpt = &client.ListOptions{Namespace: *project.Spec.Namespace}
+	} else if t.SeedName() != "" {
+		listOpt = client.MatchingFields{gardencore.ShootSeedName: t.SeedName()}
+	} else {
+		return nil, errors.New("invalid target, must have either project or seed specified for targeting a shoot")
+	}
```

Comment:
do we really want to forbid this?

Reply:
How do you mean? The "you must target a project or seed when you want to target a shoot"-logic is ingrained everywhere in the targeting.

Reply:
Already now I can target a shoot without targeting a project or seed previously
``` bash
$ gardenctl target garden landscape-dev
Successfully targeted garden "landscape-dev".
```
```bash
$ gardenctl target shoot myshoot`
`Successfully targeted shoot "myshoot".
```
and it will automagically figure out the project
```bash
cat ~/.garden/target.yaml
garden: landscape-dev
project: myproject
shoot: myshoot
```

Reply:
Okay then, I will refactor everything.

Reply:
Actually, we should do this in a separate PR. Changing this core aspect will make it really hard to review otherwise.

Reply:
ok fine with me